### PR TITLE
Add code to fix samples index

### PR DESCRIPTION
**Pull request by shreeyaharma**

### CODE REVIEW

File: assets/model_monitoring/components/genai_token_statistics/genai_token_statistics_compute_metrics/spec.yaml [hunk 1]
```diff
@@ -18,12 +18,15 @@ outputs:
...
+  token_count:
+    type: mltable
+    mode: direct
 
 code: ../../src/
 entry:
   file: ./genai_token_statistics_compute_metrics/run.py
 conf:
-  spark.hadoop.aml.enable_cache : "true"
+  #spark.hadoop.aml.enable_cache : "true"
```
Comment (yunjie-hub) [inline assets/model_monitoring/components/genai_token_statistics/genai_token_statistics_compute_metrics/spec.yaml:29]: why disable cache for this component? not ready?

Comment (yunjie-hub) [reply assets/model_monitoring/components/genai_token_statistics/genai_token_statistics_compute_metrics/spec.yaml:29]: Could you just remove it if it's not ready.

File: assets/model_monitoring/components/src/genai_token_statistics_compute_metrics/run.py [hunk 2]
```diff
@@ -4,11 +4,14 @@
 """Entry script for Computing Metrics Spark Component."""
 
 import argparse
+import os
 
+from pyspark.sql.types import IntegerType, StructField, StructType, StringType
```
Comment (yunjie-hub) [inline assets/model_monitoring/components/src/genai_token_statistics_compute_metrics/run.py:9]: do you need IntegerType?

**Review by yunjie-hub — DISMISSED** [review]

**Review by yunjie-hub — APPROVED** [review]

File: assets/model_monitoring/components/src/genai_token_statistics_compute_metrics/run.py [hunk 3]
```diff
@@ -27,8 +31,26 @@ def run():
...
+    metadata_schema = StructType(
+        [
+            StructField("metric_name", StringType(), True),
+            StructField("group", StringType(), True),
+            StructField("group_dimension", StringType(), True),
+            StructField("samples_name", StringType(), True),
+            StructField("asset", StringType(), True),
+        ]
+    )
+    sample_df = spark.createDataFrame(samples_index_rows, metadata_schema)
```
Comment (RichardLi1437) [inline assets/model_monitoring/components/src/genai_token_statistics_compute_metrics/run.py:52]: seems only the asset url is logged in this file, why do we need this? Can't we get the output url from the run metadata?

**Pull request merged**